### PR TITLE
feat: handle MsgCreateVestingAccount

### PR DESCRIPTION
**Pull request by huichiaotsou**
## Description

Closes: #XXXX

This PR will handle the MsgCreateVestingAccount as the auth module currently only handles genesis vesting accounts. 

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [x] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [x] targeted the correct branch
- [x] provided a link to the relevant issue or specification
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [x] reviewed "Files changed" and left comments if necessary
- [x] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: modules/auth/handle_msg.go [hunk 1]
```diff
@@ -14,8 +20,40 @@ func (m *Module) HandleMsg(_ int, msg sdk.Msg, tx *juno.Tx) error {
...
 		log.Error().Str("module", "auth").Err(err).
 			Str("operation", "refresh account").
-			Msgf("error while refreshing accounts after message of type %s", msg.Type())
+			Msgf("error while refreshing accounts after message of type %s", proto.MessageName(msg))
+	}
+
+	if cosmosMsg, ok := msg.(*vestingtypes.MsgCreateVestingAccount); ok {
+		err = m.handleMsgCreateVestingAccount(cosmosMsg)
+		if err != nil {
+			return fmt.Errorf("error while handling to MsgCreateVestingAccount %s", err)
```
Comment (MonikaCat) [inline modules/auth/handle_msg.go]:
```suggestion
			return fmt.Errorf("error while handling MsgCreateVestingAccount %s", err)
```

File: modules/auth/handle_msg.go [hunk 2]
```diff
@@ -14,8 +20,40 @@ func (m *Module) HandleMsg(_ int, msg sdk.Msg, tx *juno.Tx) error {
...
+
+func (m *Module) handleMsgCreateVestingAccount(msg *vestingtypes.MsgCreateVestingAccount) error {
+	va, err := convertBaseVestingAccountFromMsg(msg)
+	if err != nil {
+		return fmt.Errorf("error while converting MsgCreateVestingAccount to base vesting account %s", err)
+	}
+
+	err = m.db.StoreVestingAccountFromMsg(va)
+	if err != nil {
+		return fmt.Errorf("error while storing to base vesting account from msg %s", err)
```
Comment (MonikaCat) [inline modules/auth/handle_msg.go]:
```suggestion
		return fmt.Errorf("error while storing base vesting account from msg %s", err)
```

**Comment by MonikaCat** [comment]
@huichiaotsou which chain have you tested it with and at what height? I am testing it locally with desmos but so far it can't find any msgs with `MsgCreateVestingAccount` type

**Comment by huichiaotsou** [comment]
> @huichiaotsou which chain have you tested it with and at what height? I am testing it locally with desmos but so far it can't find any msgs with `MsgCreateVestingAccount` type

so far I know only 1 case for MsgCreatePeriodicVestingAccount which was our validator account created by Kwun on agoric, it's at [height 3507144](https://agoric.bigdipper.live/transactions/05FA8772BBB192F684B6FBD67C1607C3E37B0BFEF2FF11D44BCFDCE045810DD4)

**Comment by kwunyeung** [comment]
These accounts in this transaction are all not being handled.

https://agoric.bigdipper.live/transactions/7A3AFC2DD94A023419EC41846AE5FD0C36A070581B19623B8855CB0FE326CD0C

**Comment by MonikaCat** [comment]
Same issue as mentioned in PR #318, the `start_time` value is always null. Can you try to pass the timestamp from the tx? Other than that everything looks good and it's working correctly.

**Review by MonikaCat — APPROVED** [review]
LGTM

**Comment by rmontagnin** [comment]
Was the problem raised by Kwun solved? Is the account now being stored correctly?

**Comment by MonikaCat** [comment]
@RiccardoM Yes the issue raised by @kwunyeung was fixed in PR #318 that was directly targeting `v2/chains/agoric/mainnet` branch

**Review by rmontagnin — APPROVED** [review]

**Pull request merged**